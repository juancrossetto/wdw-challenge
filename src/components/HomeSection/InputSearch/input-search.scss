@use '../../../scss/' as *;

.input-search {
  width: 100%;
  height: 100%;
  border-radius: 10px;
  background-color: $gray-one;

  &__input {
    background-color: $gray-one !important;
    border: none !important;
  }
  &__icon {
    background-color: $gray-one;
    margin: auto;
    cursor: pointer;
    border-radius: 10px;
  }
  &__img {
    width: 30px;
    height: 30px;
    margin-right: 9px;
    @include for-phone-only {
      width: 21px;
      height: 22px;
    }
  }
}

.focus-visible {
  box-shadow: none !important;
}
