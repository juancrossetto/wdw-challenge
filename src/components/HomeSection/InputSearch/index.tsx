import React from 'react';
import { FC, InputHTMLAttributes } from 'react';
import { iconSearch } from '../../../assets/images';
import Input from '../../Input';
import './input-search.scss';

interface InputSearchProps extends InputHTMLAttributes<HTMLInputElement> {
  type?: string;
  onChange: (e: any) => void;
  width?: string;
  height?: string;
}

const InputSearch: FC<InputSearchProps> = ({ type, onChange, width, height }) => {
  return (
    <div className="input-group input-search">
      <Input
        type={type || 'text'}
        onChange={onChange}
        className="form-control bg-red input-search__input"
        style={{ borderRadius: '10px' }}
        data-testid="input-search-test"
      />
      <div className="input-group-prepend input-search__icon">
        <img src={iconSearch} alt="Search Post" className="input-search__img" />
      </div>
    </div>
  );
};

export default InputSearch;
