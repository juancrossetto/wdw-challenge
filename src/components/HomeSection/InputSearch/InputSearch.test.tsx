import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import InputSearch from '.';

describe('InputSearch component', () => {
  test('InputSearch render correctly', () => {
    const mockCallback = jest.fn();
    const tree = renderer.create(<InputSearch value={'test'} onChange={mockCallback} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('InputSearch onchange', () => {
    const mockCallback = jest.fn();
    const { getByTestId } = render(<InputSearch value={'test'} onChange={mockCallback} />);
    const input = getByTestId('input-search-test') as HTMLInputElement;
    fireEvent.change(input, { target: { value: 'test edited' } });
    expect(input.value).toBe('test edited');
  });
});
