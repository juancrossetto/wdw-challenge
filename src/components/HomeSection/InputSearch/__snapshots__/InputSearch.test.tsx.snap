// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSearch component InputSearch render correctly 1`] = `
<div
  className="input-group input-search"
>
  <input
    className="custom-input form-control bg-red input-search__input"
    data-testid="input-search-test"
    onChange={[MockFunction]}
    style={
      Object {
        "borderRadius": "10px",
      }
    }
    type="text"
  />
  <div
    className="input-group-prepend input-search__icon"
  >
    <img
      alt="Search Post"
      className="input-search__img"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
</div>
`;
